### DISCUSSION TITLE
Does _StreamingResponse correctly inherit Response

### DISCUSSION THREAD

**Discussion by whymatter** [encode/starlette#2731] [Potential Issue]
I came across this [discussion](https://github.com/encode/starlette/commit/d771bb7a5ecfcd658735cf07ba6f1dd4ac057ccd#r146823236) since we were affected by this change. While I understand that we can not expect a`StreamingResponse` I have another issue.

_Disclaimer: Since I am not a 100% Python typing expert please correct me if I am wrong but my understanding is that a derived class should have all attributed that the base class publishes._

The `_StreamingResponse` inherits `Response` but does not call the super constructur, hence `_StreamingResponse` does not have the attributes that `Response` promises (e.g. `body`). Imho there are three possible solutions:
* Make body private `_body`. This hides it from external use (I dont think that is the way to go)
* Add `body` to `_StreamingResponse` e.g. by calling `super`
* Remove `body` from Response or extract another base class like `BaseResponse` that does not have body

**Comment by adriangb** [comment]
You're right that something is wrong with our response inheritance chain. I think we should just re-design it, before 1.0. @abersheeran has thought about this before and has a proposed solution.

I would be keep to accept a PR with the fix, ideally one that does the right thing but still tries to minimize breaking changes.

**Comment by abersheeran** [comment]
If you're interested, I have a code here that I think is a good inheritance of Response class. https://github.com/abersheeran/baize/blob/master/baize/asgi/responses.py
